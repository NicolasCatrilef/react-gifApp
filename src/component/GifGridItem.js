import React from 'react'

export const GifGridItem = ( { id, title, url } ) => {
    
    // console.log( id, title, url );
    
    return (
        <div className = "card animate__animated animate__bounce">
            {/* { img.titule } */}
            <img src = { url } alt = { title } />
           <p> { title } </p> 
        </div>
    )
}
