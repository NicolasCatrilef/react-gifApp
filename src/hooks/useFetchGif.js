import { useEffect, useState } from "react"
import { getGifts } from '../helpers/getGifs'

export const useFetchGif = ( category ) => {

    const [state, setstate] = useState({
        data: [],
        loading: true
    });

    useEffect(() => {
        getGifts( category )
            .then( img => {
                setTimeout(() => {
                    setstate({
                        data: img,
                        loading: false
                    })
                }, 3000);
            } )
    }, [ category ]);


    return state;
}