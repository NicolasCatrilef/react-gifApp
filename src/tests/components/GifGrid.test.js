import { shallow } from "enzyme";
import { GifGrid } from "../../component/GifGrid";
import { useFetchGif } from "../../hooks/useFetchGif";
jest.mock( '../../hooks/useFetchGif' );

describe('Pruebas GifGrid', () => {
    
    const catg = 'Kimetsu No Yaiba';
    
    test('Renderizar GifGrid', () => {
        
        useFetchGif.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid value = { catg }/> ) ;

        expect( wrapper ).toMatchSnapshot();
    });
    

    test('Debe mostrar img con UseFetch', () => {
        
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier Cosa'
        },
        {
            id: '123',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier Cosa'
        }];
        
        useFetchGif.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow( <GifGrid value = { catg }/> ) ;

        expect( wrapper ).toMatchSnapshot();
        expect ( wrapper.find( 'p' ).exists() ).toBe( false );
        // console.log( wrapper.find( 'GifGridItem' ).length );
        expect ( wrapper.find( 'GifGridItem' ).length ).toBe( gifs.length ); 
    });
    
});
