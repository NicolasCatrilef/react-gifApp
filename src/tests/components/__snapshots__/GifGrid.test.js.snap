// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas GifGrid Debe mostrar img con UseFetch 1`] = `
<Fragment>
  <h3>
     
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier Cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Cualquier Cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas GifGrid Renderizar GifGrid 1`] = `
<Fragment>
  <h3>
     
     
  </h3>
  <p>
     Loading 
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
